File: 2-Android With Java/MVP/Day5_Threads/app/src/main/java/com/example/day5_threads/mainscreen/view/MainActivity.java
package com.example.day5_threads.mainscreen.view;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.example.day5_threads.R;
import com.example.day5_threads.allproducts.view.AllProductsActivity;
import com.example.day5_threads.favproducts.view.FavoriteProductActivity;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        Button btnMain = findViewById(R.id.Main);
        Button btnFav = findViewById(R.id.Fav);
        Button btnExit = findViewById(R.id.button3);

        btnMain.setOnClickListener(v -> {
            startActivity(new Intent(this, AllProductsActivity.class));
        });

        btnFav.setOnClickListener(v -> {
            startActivity(new Intent(this, FavoriteProductActivity.class));
        });

        btnExit.setOnClickListener(v -> {
            finish();
        });
    }
}

File: 5-Architecture design patterns/MVVM/Products/app/src/main/java/com/example/products/features/main/view/MainActivity.kt
package com.example.products.features.main.view

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.example.products.databinding.ActivityMainBinding
import com.example.products.features.allproducts.view.AllProductsActivity
import com.example.products.features.favourite.view.FavActivity


class MainActivity : AppCompatActivity() {

    lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
//        setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.Main.setOnClickListener{
            startActivity(Intent(this, AllProductsActivity::class.java))
        }

        binding.Fav.setOnClickListener{
            startActivity(Intent(this, FavActivity::class.java))
        }
        binding.button3.setOnClickListener{
            finish();
        }






    }
}